python/ray/tests/test_runtime_env.py

```diff
@@ -247,6 +247,20 @@ def f():
     assert len(errors) == 0
 
 
+@pytest.mark.skipif(sys.platform == "win32", reason="pip not supported on Windows.")
+def test_failed_job_env_no_hang():
+    """Test that after a failed job-level env, tasks can still be run."""
+    ray.init(runtime_env={"pip": ["ray-doesnotexist-123"]})
+
+    @ray.remote
+    def f():
+        import pip_install_test
+
+        return True
+
+    assert ray.get(f.options(runtime_env={"pip": ["pip-install-test==0.5"]}).remote())
```

Comment:
Actually I also would like to test the case where runtime env is not given (so it uses the default job runtime env).

Can you test this case?